lib/api/interceptor.ts

```diff
@@ -0,0 +1,48 @@
+function _getRulesFromStorage() {
+  return JSON.parse(localStorage.getItem(`${ location.origin }-shredder`))
+}
+
+function _checkParamsEquality(requestBody = {}, responseParams) {
+  const requestParams = JSON.parse(requestBody);
+  let isEqual = true;
+
+  Object.keys(requestParams).forEach((key) => {
+    const regex = responseParams[key] ? new RegExp(responseParams[key]) : null;
+
+    if (regex && !requestParams[key].match(regex)) {
+      isEqual = false;
+    }
+  })
+
+  return isEqual;
+}
+
+function _createResponse(cb, responseObject) {
+  const response = {
+    status: responseObject.statusCode || 200,
+    data: JSON.stringify(responseObject.body || {}),
+    headers: responseObject.headers || {}
+  };
+
+  if (responseObject.delay > 0) {
+    return setTimeout(() => cb(response), responseObject.delay);
+  }
+
+  return cb(response);
+}
+
+const rules = _getRulesFromStorage();
+
+xhook.before((request, responder) => {
```

Comment:
There aren't any exports here.

This should be turned into a class that allows installation/uninstallation of the interceptor for when we don't want to intercept anything.

We should also check if the rule is enabled (should it run).